client-v2/src/lib/guration/DedupeNode.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import React from 'react';
+import { DedupeContext } from './Context';
+import GetDuplicate from './GetDuplicate';
+import type { DedupeContextType } from './Context';
+import type { Path } from './utils/path';
+
+type ExternalDedupeNodeProps = {|
+  index: number,
+  path: Path[],
+  dedupeKey?: string,
+  type: string,
+  children: *
+|};
+
+type DedupeNodeProps = {|
+  ...ExternalDedupeNodeProps,
+  context: DedupeContextType
+|};
+
+class DedupeNode extends React.Component<DedupeNodeProps> {
+  componentDidMount = () => this.reregister();
+  componentDidUpdate = () => this.reregister();
+  componentWillUnmount = () => this.deregister();
+
+  get dedupeContext() {
```

Comment:
What is the `get` keyword doing here?

Reply:
It means I can drop the parens when calling this method (as it has arity 0). Someone once told me about the "uniform access principle" and I ate it right up! Also it means I can destructure like below and it runs the getter. I only use them if they're not side effecting.

```js
const { dedupeContext } = this;
```